packages/client/src/runtime/core/init/resolveDatasourceUrl.ts

```diff
@@ -29,12 +31,12 @@ export function resolveDatasourceUrl({
 
   // env var is set for use but url is undefined
   if (datasourceUrl?.fromEnvVar !== undefined && resolvedUrl === undefined) {
-    if (TARGET_ENGINE_TYPE === 'edge') {
+    if (TARGET_ENGINE_TYPE === 'edge' && detectRuntime() === 'workerd') {
       throw new PrismaClientInitializationError(
         `error: Environment variable not found: ${datasourceUrl.fromEnvVar}.
-In edge runtimes, only \`process.env\` & \`globalThis\` are read, not \`.env\`.
 
-To solve this, provide the URL directly: https://pris.ly/d/datasource-url`,
+In Cloudflare Module Workers, the URL is in the Worker's \`env\` parameter of \`fetch\`.
+To solve this, provide the URL directly: https://pris.ly/d/cloudflare-datasource-url`,
```

Comment:
```suggestion
In Cloudflare module Workers, environment variables are available only in the Worker's \`env\` parameter of \`fetch\`.
To solve this, provide the connection string directly: https://pris.ly/d/cloudflare-datasource-url`,
```

Reply:
I'm a bit puzzled by the terminology "Cloudflare module Workers" here.
"Cloudflare Workers (ES Format)" sounds a bit more like it: https://developers.cloudflare.com/workers/learning/migrate-to-module-workers/.